core/src/main/java/org/geysermc/geyser/network/UpstreamPacketHandler.java

```diff
@@ -249,29 +248,17 @@ public PacketSignal handle(ModalFormResponsePacket packet) {
     }
 
     private boolean couldLoginUserByName(String bedrockUsername) {
-        if (geyser.getConfig().getSavedUserLogins().contains(bedrockUsername)) {
+        if (geyser.getConfig().getSavedUserLogins() != null && geyser.getConfig().getSavedUserLogins().contains(bedrockUsername)) {
```

Comment:
If I recall correctly, this should never be null?

Reply:
It unfortunately is when the config's list is empty with no entries :/

Reply:
Then why is it only an issue now?

Reply:
As far as I can tell it was an issue before too, just not one users would run into - we add two dummy entries there.
Only discovered this by accident when testing

Reply:
Then instead can we default the field to `Collections.emptyList()`?

Reply:
Hm, looks like we already do?
https://github.com/GeyserMC/Geyser/blob/master/core/src/main/java/org/geysermc/geyser/configuration/GeyserJacksonConfiguration.java#L62-L63

Reply:
is this a Jackson moment? 👀

Reply:
Yup
Looks like you can change that behavior with an annotation: `@JsonSetter(nulls = Nulls.AS_EMPTY)` 
source:
https://stackoverflow.com/questions/32398885/jackson-deserializer-change-null-collection-to-empty-one

Reply:
Setting that value globally on the ObjectMapper as described in that link would probably be best

Reply:
fixed by changing the deserializer settings: [source](https://stackoverflow.com/questions/32398885/jackson-deserializer-change-null-collection-to-empty-one)
```java
ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory())
                .setDefaultSetterInfo(JsonSetter.Value.forValueNulls(Nulls.AS_EMPTY));
                ```